addons/account/models/account_journal_dashboard.py

```diff
@@ -701,8 +701,8 @@ def _get_journal_dashboard_bank_running_balance(self):
                              JOIN account_move move ON move.id = stl.move_id
                             WHERE stl.statement_id IS NULL
                               AND move.state != 'cancel'
-                              AND move.journal_id = journal.id
-                              AND move.company_id = ANY(%s)
+                              AND stl.journal_id = journal.id
+                              AND stl.company_id = ANY(%s)
```

Comment:
This diff is the heart of this PR, as it's the last big bottleneck in the accounting dashboard.
It basically converts the expensive multi-batch Hash Join (heavy on IO) into a Nested loop with the Pkey of `account.move`.
It's ok to leave the condition on the move.state, because of the `LIMIT 1`.

This diff is an order of magnitude difference on a staging (450ms -> 45ms). On production, the impact will probably be more significant due to the reduced IO R/W.